Documentation/upgrade.md

```diff
@@ -0,0 +1,293 @@
+Rook is a distributed software system and therefore there are multiple components to individually upgrade in the sequence defined in this guide.
+After each component is upgraded, it is important to verify that the cluster returns to a healthy and fully functional state.
+
+
+This guide is just the beginning of upgrade support in Rook.
+The goal is to provide prescriptive guidance and knowledge on how to upgrade a live Rook cluster and we hope to get valuable feedback from the community that will be incorporated into an automated upgrade solution by the Rook operator.
+
+We welcome feedback and opening issues!
+
+## Considerations
+With this manual upgrade guide, there are a few notes to consider:
+* **WARNING:** Upgrading a Rook cluster is a manual process in its very early stages.  There may be unexpected issues or obstacles that damage the integrity and health of your storage cluster, including data loss.  Only proceed with this guide if you are comfortable with that.
+* Rook is still in an alpha state.  Migrations and general support for breaking changes across versions are not supported or covered in this guide.
+
+## Prerequisites
+In order to successfully upgrade a Rook cluster, the following prerequisites must be met:
+* The cluster should be in a healthy state with full functionality.
+Review the [health verification section](#health-verification) in order to verify your cluster is in a good starting state.
+* `dataDirHostPath` must be set in your Cluster spec.
+This persists metadata on host nodes, enabling pods to be terminated during the upgrade and for new pods to be created in their place.
+More details about `dataDirHostPath` can be found in the [Cluster CRD readme](./cluster-crd.md#cluster-settings).
+
+The minimal sample Cluster spec that will be used in this guide can be found below (note that the specific configuration may not be applicable to all environments):
+```
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: rook
+---
+apiVersion: rook.io/v1alpha1
+kind: Cluster
+metadata:
+  name: rook
+  namespace: rook
+spec:
+  versionTag: v0.5.0
+  dataDirHostPath: /var/lib/rook
+  storage:
+    useAllNodes: true
+    useAllDevices: true
+    storeConfig:
+      storeType: bluestore
+      databaseSizeMB: 1024
+      journalSizeMB: 1024
+```
+
+## Health Verification
+Before we begin the upgrade process, let's first review some ways that you can verify the health of your cluster, ensuring that the upgrade is going smoothly after each step.
+Most of the health verification checks for your cluster during the upgrade process can be performed with the Rook toolbox.
+For more information about how to run the toolbox, please visit the [Rook toolbox readme](./toolbox.md#running-the-toolbox-in-kubernetes).
+
+### Pods all Running
+In a healthy Rook cluster, the operator and all Rook namespace pods should be in the `Running` state and have zero pod restarts.
```

Comment:
why is the zero pod restarts critical? They won't prevent the upgrade.

Reply:
it's not critical, but having pod restarts usually doesn't indicate a healthy cluster.  i'll clarify the wording here to no longer imply that 0 is critical.